cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -84,28 +84,36 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
 
   @Override
   public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal) throws Exception {
-    LOG.debug("Checking visibility of {} for principal {}.", entityIds, principal);
-    return authorizerInstantiator.get().isVisible(entityIds, principal);
+    if (!isSecurityAuthorizationEnabled()) {
+      return entityIds;
+    }
+
+    Set<EntityId> results = new HashSet<>();
+    // filter out entity id which is in system namespace and principal is the master user
+    for (EntityId entityId : entityIds) {
+      if (bypassCheck(entityId, principal)) {
```

Comment:
To improve readability, let's call this `isSystemEntity()` or something similar.